packages/fluentui/react-bindings/src/hooks/useTrigger.tsx

```diff
@@ -0,0 +1,86 @@
+import { Ref } from '@fluentui/react-component-ref';
+import * as React from 'react';
+import * as ReactIs from 'react-is';
+
+import { childrenExist } from '../utils/childrenExist';
+
+type UseTriggerOptions = {
+  children?: React.ReactNode;
+  trigger?: React.ReactElement;
+};
+
+// https://github.com/facebook/react/blob/c4e0768d7487a9359b74986e3b07841d2520f593/packages/react-dom/src/events/getListener.js#L15-L22
+function isDisabledInteractive(node: Node): boolean {
+  return (
+    (node.nodeName === 'BUTTON' ||
+      node.nodeName === 'INPUT' ||
+      node.nodeName === 'SELECT' ||
+      node.nodeName === 'TEXTAREA') &&
+    (node as HTMLButtonElement).disabled
+  );
+}
+
+function isInteractiveFilter(node: Node) {
+  return isDisabledInteractive(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
+}
+
+export function useTrigger(props: UseTriggerOptions): React.ReactElement | null {
```

Comment:
Comments would be appreciated about what the use case is for this.

Reply:
```suggestion
/**
 * Performs consistent checks for components that are using `trigger` pattern (`Tooltip`, `Popup`). Ensures that
 * `children` or `trigger` props will always pass a valid React element to what additional props and handlers can 
 * be applied.  
 */
```

Does it makes sense?

Reply:
Should this be `useTriggerElement` then? Pass in children/trigger element, get back the element which should be used as the trigger element.

Reply:
Minor modification, adjust to indicate what the ins and outs are, like:

> Given the `children` and expected element to `trigger` on, returns the correct trigger element. Performance consistent checks for components that are using the `trigger` pattern (etc)

That way the user knows generally how to use it. (They can look at the examples you mentioned for more details 👍 )

Otherwise looks good